jdeeco-core/src/cz/cuni/mff/d3s/deeco/runtimelog/RuntimeLogger.java

```diff
@@ -0,0 +1,537 @@
+	 */
+	public static void log(RuntimeLogRecord record) throws IOException,
+			IllegalStateException {
+		log(record, false);
+	}
+
+	/**
+	 * Writes the data from the given <em>snapshotData</em> argument to the {@link RuntimeLogger#dataWriter}.
+	 * The written record has a format of an <a href="http://en.wikipedia.org/wiki/XML">XML</a>
+	 * element. The values from the given <em>snapshotData</em> argument are stored as attributes of the
+	 * written <a href="http://en.wikipedia.org/wiki/XML">XML</a> element. Also an index record
+	 * pointing to this snapshot record is written using the {@link RuntimeLogger#indexWriter}.
+	 * <p> The {@link Writer}s are being flushed when this method is called. </p>
+	 * @param snapshotData contains the data that will be logged.
+	 * @throws IOException Thrown if there is a problem writing into the {@link Writer}s
+	 * for the logging.
+	 * @throws IllegalStateException Thrown if the {@link RuntimeLogger#init} method hasn't
+	 * been called before this method is being called.
+	 * @throws IllegalArgumentException Thrown if the given argument is null.
+	 */
+	public static void logSnapshot(RuntimeLogRecord snapshotData)
+			throws IOException, IllegalStateException {
+		log(snapshotData, true);
+	}
+
+	/**
+	 * Writes the data from the given <em>record</em> to the {@link RuntimeLogger#dataWriter}.
+	 * The written record has a format of an <a href="http://en.wikipedia.org/wiki/XML">XML</a>
+	 * element. The values from the given <em>record</em> argument are stored as attributes of the
+	 * written <a href="http://en.wikipedia.org/wiki/XML">XML</a> element. If the <em>snapshot</em>
+	 * argument is true also an index record pointing to the current snapshot record is written
+	 * using the {@link RuntimeLogger#indexWriter}.
+	 * <p> The {@link Writer}s are being flushed when the <em>snapshot</em> argument is true. </p> 
+	 * @param record contains the data that will be logged.
+	 * @param snapshot indicates whether the <em>record</em> holds a snapshot (true) or an event (false).
+	 * @throws IOException Thrown if there is a problem writing into the {@link Writer}s
+	 * for the logging.
+	 * @throws IllegalStateException Thrown if the {@link RuntimeLogger#init} method hasn't
+	 * been called before this method is being called.
+	 * @throws IllegalArgumentException Thrown if the given argument is null.
+	 */
+	private static void log(RuntimeLogRecord record, boolean snapshot) // TODO: check whether the record fields are not null
+			throws IOException, IllegalStateException {
+		if (instance.timeProvider == null)
+			throw new IllegalStateException(
+					"The SimLogger class not initialized.");
+		if (record == null)
+			throw new IllegalArgumentException(String.format(
+					"The argument \"%s\" is null.", "record"));
+
+		StringBuilder recordBuilder = new StringBuilder();
+		String recordName = snapshot ? SNAPSHOT_RECORD_NAME : EVENT_RECORD_NAME;
```

Comment:
I would not distinguish between snapshot record name and event record name. The difference between event or snapshot is anyway in the record type.